src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.h

```diff
@@ -60,6 +60,7 @@ class ChaseMovementGenerator : public MovementGeneratorMedium<T, ChaseMovementGe
 
 private:
     TimeTrackerSmall i_leashExtensionTimer;
+    TimeTrackerSmall r_leashExtensionTimer;
```

Comment:
Why are you using `r_` prefix here? I suppose `i_` stands for `internal`.
So if you want a special timer for ranged attacks, then call it in some other way, like `i_rangedLeashExtensionTimer` or something.